Makefile

```diff
@@ -50,8 +48,6 @@ $(KUBEADDONS_TEST_KUBECONFIG): install-bin
 .PHONY: kind-test
 kind-test: create-kind-cluster
 	KUBEADDONS_REPO=$(KUBEADDONS_REPO) TESTING_BRANCH=$(TESTING_BRANCH) KUBEADDONS_TEST_KUBECONFIG=$(KUBEADDONS_TEST_KUBECONFIG) $(ROOT_DIR)/run-tests.sh
-	KUBECONFIG=$(KUBEADDONS_TEST_KUBECONFIG) bin/kind delete cluster
```

Comment:
Why was this (cleanup) not required?

Reply:
I gave LGTM thinking, there is a good reason for this removal, but not obvious to me.

Please share the details before merge.

Reply:
This stuff runs in the make clean goal